src/workloads/query/LargeProjections.yml

```diff
@@ -0,0 +1,204 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: >
+  This workload runs collscan queries with a large projection on around 20 fields.
+
+Keywords:
+- Loader
+- CrudActor
+- QuiesceActor
+- insert
+- find
+
+GlobalDefaults:
+  Database: &Database test
+  Collection: &Collection Collection0
+  DocumentCount: &DocumentCount 1e6
+  Repeat: &Repeat 50
+  Threads: &Threads 1
+  MaxPhases: &MaxPhases 9
+
+Actors:
+# Clear any pre-existing collection state.
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+# Insert documents with more than 20 fields for projection. The documents also contain 20 different
+# nested field path for projection.  This test targets workloads running queries with large
+# projections (inclusive, exclusive, dotted-path, and mixed).
+- Name: InsertDocumentWithManyFields
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document: {
```

Comment:
It might be more valuable from a testing perspective if some of these fields had different types/were randomized. There are some examples in other workloads, e.g. [here](https://github.com/mongodb/genny/blob/12953ac9a76cea4de4becdddbcc8d8da4ea887cb/src/phases/query/CollScanComplexPredicate.yml#L279).